packages/tools/src/utilities/segmentation/VolumetricCalculator.ts

```diff
@@ -1,79 +1,188 @@
 import type { Types } from '@cornerstonejs/core';
 import type { NamedStatistics } from '../../types';
-import { BasicStatsCalculator } from '../math/basic';
+import {
+  BasicStatsCalculator,
+  InstanceBasicStatsCalculator,
+} from '../math/basic/BasicStatsCalculator';
 
 const TEST_MAX_LOCATIONS = 10;
 
+interface VolumetricState {
+  maxIJKs: Array<{
+    value: number;
+    pointLPS?: Types.Point3;
+    pointIJK?: Types.Point3;
+  }>;
+}
+
+function createVolumetricState(): VolumetricState {
+  return {
+    maxIJKs: [],
+  };
+}
+
+function volumetricStatsCallback(
+  state: VolumetricState,
+  data: {
+    value: number | Types.RGB;
+    pointLPS?: Types.Point3;
+    pointIJK?: Types.Point3;
+  }
+): void {
+  const { value } = data;
+  const { maxIJKs } = state;
+  const length = maxIJKs.length;
+
+  if (
+    typeof value !== 'number' ||
+    (length >= TEST_MAX_LOCATIONS && value < maxIJKs[0].value)
+  ) {
+    return;
+  }
+
+  if (!length || value >= maxIJKs[length - 1].value) {
+    maxIJKs.push(
+      data as {
+        value: number;
+        pointLPS?: Types.Point3;
+        pointIJK?: Types.Point3;
+      }
+    );
+  } else {
+    for (let i = 0; i < length; i++) {
+      if (value <= maxIJKs[i].value) {
+        maxIJKs.splice(
+          i,
+          0,
+          data as {
+            value: number;
+            pointLPS?: Types.Point3;
+            pointIJK?: Types.Point3;
+          }
+        );
+        break;
+      }
+    }
+  }
+
+  if (maxIJKs.length >= TEST_MAX_LOCATIONS) {
+    maxIJKs.splice(0, 1);
+  }
+}
+
+function volumetricGetStatistics(
+  state: VolumetricState,
+  stats: NamedStatistics,
+  options: {
+    spacing?: number[] | number;
+    unit?: string;
+  }
+): NamedStatistics {
+  const { spacing } = options;
+  const volumeUnit = spacing ? 'mm³' : 'voxels³';
```

Comment:
Please add methods to getCalibratedLengthUnitsAndScale to get the appropriate volume unit.  Also, voxel is the unit, voxels^3 would be a 9 dimensional hyper cube.